### PR TITLE
*: Rename kuksa-viss-client kuksa-client

### LINKED ISSUE #318: Rename kuksa-viss-client to kuksa-client
KUKSA currently supports VISS and GRPC, to reflect that, we should rename it. 
for this issue, rename the package/folder here. make sure testclient works, and CI still builds.
No need to adapt feeders

### PR DESCRIPTION
**Pull request by romainletendart**
Package directory and name shall use a dash as per https://pep8.org/#package-and-module-names.
Entry point shall also use a dash as it is common for UNIX programs. The imported python package has to use an underscore however otherwise we cannot import it (SyntaxError).

Fixes [#318](https://github.com/eclipse/kuksa.val/issues/318).

### CODE REVIEW

File: data/vss-core/README.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ VERBOSE: SubscriptionHandler::publishForVSSPath: set value 8.927633333 for path
 VERBOSE: SubscriptionHandler::publishForVSSPath: set value 12.244000434875488 for path Vehicle.Speed
 ```
 
-The [VISS client](https://github.com/eclipse/kuksa.val/blob/master/kuksa_viss_client/README.md) can be used to verify that data actually is correctly interpreted.
+The [VISS client](https://github.com/eclipse/kuksa.val/blob/master/kuksa-client/README.md) can be used to verify that data actually is correctly interpreted.
```
Comment (erikbosch) [inline data/vss-core/README.md]: Do a file search for VISS, it shall only remain in those places where we refer to the VISS standard or the the VISS (websocket) protocol. (And in the latter cases we should possibly state that it is the websocket variant of VISS we support, not HTPPS/MQTT)

File: doc/pictures/testclient_basic.cast [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
-[0.36832, "o", "python -m kuksa_viss_client"]
-[0.368546, "o", "\u001b[27D\u001b[38;5;26mpython\u001b[38;5;0m\u001b[m\u001b(B \u001b[38;5;39m-m\u001b[38;5;0m\u001b[m\u001b(B \u001b[38;5;39mkuksa_viss_client\u001b[38;5;0m\u001b[m\u001b(B"]
+[0.36832, "o", "python -m kuksa_client"]
+[0.368546, "o", "\u001b[27D\u001b[38;5;26mpython\u001b[38;5;0m\u001b[m\u001b(B \u001b[38;5;39m-m\u001b[38;5;0m\u001b[m\u001b(B \u001b[38;5;39mkuksa_client\u001b[38;5;0m\u001b[m\u001b(B"]
 [1.174898, "o", "\r\n"]
 [1.175056, "o", "\u001b[38;5;0m\u001b[m\u001b(B"]
 [1.175185, "o", "\u001b[2 q"]
 [1.175268, "o", "\u001b[?2004l"]
 [1.175686, "o", "\u001b]0;python /hom"]
 [1.175759, "o", "e/wenwen/tools/kuksa.val/kuksa_viss_clien"]
```
Comment (erikbosch) [inline doc/pictures/testclient_basic.cast]: viss still appears here and there

(By the way, in the long run we should preferably regenerate/recreate this one, as I think the content is outdated, and we should preferably not include "real usernames" (like wenwen)), but that is for a separate ticket - could possibly be checked by @lukasmittag

File: kuksa-client/README.md [hunk 3]
```diff
@@ -1,46 +1,46 @@
-# Kuksa VISS Client
+# kuksa-client
```
Comment (erikbosch) [inline kuksa-client/README.md]: You mention a little about naming conventions in the PR description, e.g. when we need dash and when we need underscore. This is a third case - when referring to the components in documentation. I am thinking if we need to describe our naming strategy or a summary of them somewhere, like a table on top level README

```
Component    | Folder       | Python package name
----------------------------------------------------
Kuksa Client | kuksa-client | kuksa_client
```
I do not have a strong opinion if we always to refer to the kuksa client as "kuksa-client" or as "Kuksa Client", as long as we are consistent. After we have adapated kuksa-client we may want to synch other kuksa.val components as well, as naming today is quite inconsistent

**Comment by erikbosch** [comment]
@romainletendart - We have a build check in https://github.com/eclipse/kuksa.val/blob/master/.github/workflows/kuksa_viss_client.yml but it appears that it require a "self-hosted" builder which is not configured for boschglobal. So we will later need to move the PR to the official repo to get the CI running, to see that there is no regressions

File: doc/pictures/testclient_basic.cast (hunk 2 above)
Comment (lukasmittag) [reply doc/pictures/testclient_basic.cast]: With export PS1='name to display ' you can set what is displayed as name in terminal. I would suggest something like kuksa~. To highlight the name in a specific color you have to add export PS1='\[\033[01;31m\]\kuksa~\[\033[00m\]' for red for example

Comment (lukasmittag) [reply doc/pictures/testclient_basic.cast]: I will have a further look in how the gifs are generated (and will add some for the grpc testclient).

File: data/vss-core/README.md (hunk 1 above)
Comment (romainletendart) [reply data/vss-core/README.md]: Should I also touch the go client then as it also refers to VISS?

Comment (erikbosch) [reply data/vss-core/README.md]: I think it is best to ask @SebastianSchildt. Right now the Go-client only supports VISS (and not gRPC), so in one way it is correct to call it a VISS client. Maybe we should wait touching the internals of the Go client, until we have decided what to with it.

Comment (romainletendart) [reply data/vss-core/README.md]: OK. There is one occurrence of "GRPC" in the go client : https://github.com/eclipse/kuksa.val/blob/590198a35de7b2201bdd913750157bb9778a5214/kuksa_go_client/kuksa_viss/config.go but there is no actual implementation of a gRPC client indeed. I will leave it aside for now.

File: kuksa-client/README.md (hunk 3 above)
Comment (romainletendart) [reply kuksa-client/README.md]: Agreed for "Kuksa Client". I'm using this as a component name now.

**Comment by romainletendart** [comment]
Replaced with https://github.com/boschglobal/kuksa.val/pull/9.

**Pull request closed without merging**